### DISCUSSION TITLE
Meaning of "usage" in model capabilities?

### DISCUSSION THREAD

**Discussion by thiswillbeyourgithub** [open-webui/open-webui#4695] [Q&A] [answered]
Hi, very happy user for some time now but I have no idea what the "usage" checkbox does to an LLM. Can anyone shed some light?
Thanks

**Comment by Jonseed** [comment]
I was wondering the same thing. Does this provide API token usage statistics somewhere?

**Reply by thiswillbeyourgithub** [reply]
I still have no idea btw.

**Reply by Jonseed** [reply]
Anything change if you enable it?

**Comment by Jonseed** [comment]
@thiswillbeyourgithub I found that enabling "usage" does return some stats. It adds a `Generation Info` button below the response with three pieces of info: the number of prompt tokens, number of completion tokens, and total tokens used for the exchange.

![Screenshot 2024-09-09 220255](https://github.com/user-attachments/assets/86b22a8a-afcb-40fd-ba1d-9109ff88dba7)

**Reply by thiswillbeyourgithub** [reply]
Thanks for the followup and good idea.

Unfortunately on my side I don't see this even with the checkbox enabled:
![image](https://github.com/user-attachments/assets/a8747bda-1d2d-466d-8d35-b09575b9a7f0)


Also, I don't get at all why that would end up in the "capabilities" section

**Reply by cheahjs** [reply, marked as answer]
Enabling `Usage` sends [`{"stream_options": {"include_usage": true}}`](https://platform.openai.com/docs/api-reference/chat/create#chat-create-stream_options) with the chat completion request.

This isn't enabled by default because some OpenAI-compatible endpoints will return errors due to unknown parameters. Some endpoints will also return `usage` info regardless, at which point it'll still be shown in the UI without `Usage` being checked.

**Reply by thiswillbeyourgithub** [reply]
Ohh thank you very much. I still think that it shouldn't be in capabilities and shouldn't be simply called Usage.

**Reply by jmtatsch** [reply]
include_usage now works with ollama and vllm, maybe time to reconsider doing this as optional?
